src/applications/MachineShopSimulator.java

```diff
@@ -29,15 +29,7 @@ boolean moveToNextMachine(Job theJob, SimulationResults simulationResults) {
             return false;
         } else {// theJob has a next task
                 // get machine for next task
-            int p = ((Task) theJob.getTaskQ().getFrontElement()).getMachine();
-            // put on machine p's wait queue
-            machine[p].getJobQ().put(theJob);
-            theJob.setArrivalTime(timeNow);
-            // if p idle, schedule immediately
-            if (eList.nextEventTime(p) == largeTime) {// machine is idle
-                changeState(p);
-            }
-            return true;
+            return theJob.scheduleTask(this);

```

Comment:
If you're passing `this` as an argument, it feels like that's a sign that the logic's gone upside-down somewhere.  It might be better to just pass in the fields that the function would need?  Although, that's a lot of them.  Perhaps this is the best way of handling things... there isn't an obvious elegant solution.